webui/react/src/components/ResourceAllocationModal.tsx

```diff
@@ -0,0 +1,134 @@
+import Button from 'hew/Button';
+import Link from 'hew/Link';
+import { Modal } from 'hew/Modal';
+import { Loadable } from 'hew/utils/loadable';
+import { useObservable } from 'micro-observables';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import NodeElement from 'pages/ResourcePool/NodeElement';
+import { handlePath, paths } from 'routes/utils';
+import clusterStore from 'stores/cluster';
+import { Agent, Resource } from 'types';
+import { AnyMouseEvent } from 'utils/routes';
+
+import css from './ResourceAllocationModal.module.scss';
+
+interface Node {
+  nodeName: string;
+  slotsIds: Resource[];
+}
+
+export interface Props {
+  rpName: string;
+  accelerator: string;
+  nodes: Node[];
+  onClose: () => void;
+  isRunning: boolean;
+}
+
+interface InfoContainerProps {
+  label: string;
+  info?: JSX.Element | string;
+}
+
+const InfoContainer: React.FC<InfoContainerProps> = ({ label, info }) => {
+  return (
+    <div className={css.infoContainer}>
+      <span>{label}</span>
+      <span className={css.dots} />
+      <span>{info}</span>
+    </div>
+  );
+};
+
+const ResourceAllocationModalComponent: React.FC<Props> = ({
+  rpName,
+  nodes,
+  accelerator,
+  isRunning,
+  onClose,
+}: Props) => {
+  const [limitSlots, setLimitSlots] = useState(0);
+  const [activeNodes, setActiveNodes] = useState<Agent[]>([]);
+  const agents = Loadable.waitFor(useObservable(clusterStore.agents));
+  const footer = (
+    <div className={css.footer}>
+      <Button onClick={onClose}>Done</Button>
+    </div>
+  );
+  const renderNodes = useMemo(() => {
```

Comment:
i think this can move to inside modal.

also, items should be aligned equally

<img width="702" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/3c672c67-0b5c-4cd4-9474-d1bc83fdba66">

Reply:
> also, items should be aligned equally

this is not fixed. still aligned left